src/mumble/Messages.cpp

```diff
@@ -59,6 +59,26 @@
 		return; \
 	}
 
+/// This message is being received after the client send a Capabilities message with its preferred protocols inside.
+/// The server replies with a Capabilities message with only one protocol in it to indicate the protocol it has chosen.
+///
+void MainWindow::msgCapabilities(const MumbleProto::Capabilities &msg) {
+	if (msg.supported_protocols_size() == 0)
+		return;
```

Comment:
I think in this case the client can disconnect immediately.

Reply:
As one can see from other parts of the Messages.cpp, mumble will typically just ignore malformed messages.

Reply:
It usually does, but the crypto stuff is essential for everything to work (in contrast to e.g. a text message or someone wanting to switch their channel).

If a client gets disconnected it's clear that something went wrong whereas if it fails to decrypt messages (or even produces random noise), folks will start complaining here that "Mumble is broken" :shrug:

Reply:
Actually, if the decryption fails, the client and server will perform the _resync_ by sending CryptState message. And, the other fact is a client or server will ignore malformed CryptState message in our current implementation.

Do you think we need to change them as well or just keep this behavior consistent?

Reply:
Hm... Imo the client should get disconnected but of an invalid Crypt message is also silently ignored, I guess for the sake of consistency we can do the same here...